models/transmission.go

```diff
@@ -59,6 +60,74 @@ func (t Transmission) MarshalJSON() ([]byte, error) {
 	return json.Marshal(test)
 }
 
+// UnmarshalJSON implements the Unmarshaler interface for the Transmission type
+func (t *Transmission) UnmarshalJSON(data []byte) error {
+	var err error
+	type Alias struct {
+		Timestamps
+		ID           *string              `json:"id"`
+		Notification Notification         `json:"notification,omitempty"`
+		Receiver     *string              `json:"receiver,omitempty"`
+		Channel      Channel              `json:"channel,omitempty"`
+		Status       TransmissionStatus   `json:"status,omitempty"`
+		ResendCount  int                  `json:"resendcount"`
+		Records      []TransmissionRecord `json:"records,omitempty"`
+	}
+	a := Alias{}
+	// Error with unmarshaling
+	if err = json.Unmarshal(data, &a); err != nil {
+		return err
+	}
+
+	// Nillable fields
+	if a.ID != nil {
+		t.ID = *a.ID
+	}
+	t.Notification = a.Notification
+
+	if a.Receiver != nil {
+		t.Receiver = *a.Receiver
+	}
+	t.Channel = a.Channel
+	t.Status = a.Status
```

Comment:
If you're going to go down this validation path, then the value of Status should also be checked to ensure it's in the range of acceptable values. We have an example of that if you look into how OperatingState and AdminState are validated on the Device
https://github.com/edgexfoundry/go-mod-core-contracts/blob/master/models/device.go

Reply:
I think Notification, Channel and Channel's ChannelType would also need to have a Validate method and unit tests. Didn't fully grasp the volume of the required work but I can't get properly tested support services without this. Should these be in a different PR?

Reply:
`Validate the NotificationStatus` as part of this PR and then create an issue for the follow-up validation of the dependent types. Once the issue is created, let me know and I can assign to you